functions

```diff
@@ -30,6 +30,7 @@ emergency_shell() {
 detect_virt() {
    # Detect LXC (and other) containers
    [ -z "${container+x}" ] || export VIRTUALIZATION=1
+   [ -r /proc/1/environ ] && grep -qa "container=lx" /proc/1/environ && export VIRTUALIZATION=1
```

Comment:
if I'm reading it correctly, L32 just checks to see if the `container` environment variable is set. This doesn't happen (anymore) in recent versions of LXD, though my quick investigation suggests it was once set within containers.

Reply:
```
seth@hydrogen:~ $ [ -z "${container+x}" ] || echo "VIRTUAL"
seth@hydrogen:~ $
```
(hydrogen is a container.)

Reply:
This is an LXD container, not a VM:

```
+----------+---------+----------------------+------+-----------+-----------+
|   NAME   |  STATE  |         IPV4         | IPV6 |   TYPE    | SNAPSHOTS |
+----------+---------+----------------------+------+-----------+-----------+
| hydrogen | RUNNING | xx.xx.xxx.xxx (eth0) |      | CONTAINER | 0         |
+----------+---------+----------------------+------+-----------+-----------+
```

And yes, you do get that for PID=1 (that is, it's in `/proc/1/environ`), but for whatever reason it's not showing up when runit starts the process. Let me test something else here.

Reply:
nope. I created a runit service called `testseth`:
```
#!/bin/sh
/sbin/env > /tmp/testenv
sleep 1000000
```

output of `/tmp/testenv` is
```
PATH=/usr/bin:/usr/sbin
PWD=/etc/sv.local/testseth
```

So it doesn't look like the `container=lxc` environment variable is being passed to the runit environment (where it's being checked in the `run` script).

Reply:
> Are you 100% sure this is not an LXD VM? When/why did they change this?

This was a guess, and it turns out that it's probably wrong since /proc/1/environ shows that variable. My current thinking is that the variable is not being passed into the `run` command of the service definition, so we can't test for it. I don't know whether `/etc/runit/functions/test_virt` gets run before any services start; perhaps it should. But if it's run within a service's `run` script, the `container` variable is not found.

Reply:
Further evidence that this is not making it from runit (PID1) to the run script:
```
root         1     0  0 13:58 ?        00:00:00 runit
root        38     1  0 13:58 ?        00:00:00 runsvdir -P /run/runit/runsvdir/current
...
root      4631    38  0 15:10 ?        00:00:00 runsv docker
...
root      6977    38  0 15:22 ?        00:00:00 runsv testseth
```
```
bash-5.1# cat /proc/38/environ 
PATH=/usr/bin:/usr/sbinbash-5.1# 
```

Reply:
Found the issue with some help from gurus:

`/etc/runit/1` sets `VIRTUALIZATION` - this is confirmed. Then `/etc/runit/2` calls `svrundir` but it blanks the environment out before running it. This is why `svrundir` can't find `VIRTUALIZATION`.

So, I think the quickest and most correct thing to do is pass `VIRTUALIZATION` into the `svrundir` call in `/etc/runit/2`.

Reply:
I'm happy to bikeshed the name, but I'd really like to get this fix in soon-ish if possible. See #102 for a first cut. We can easily change it there so that `VIRTUALIZATION` only appears in `runit/1`.

Reply:
Also - I think that if we (you) decide to adopt the changes in #45, then there should be a consistent way of checking for this variable. Right now we're assigning a numeric value to it, but all the checks I've seen are checking for non-zero length. That is, `VIRTUALIZATION=0` will evaluate to true in most of the checks. This is probably not what we want, especially if we're calling it `IS_CONTAINER` which directly implies a boolean.